sympy/combinatorics/free_group.py

```diff
@@ -0,0 +1,872 @@
+from __future__ import print_function, division
+
+from sympy.core.basic import Basic
+from sympy.core.compatibility import as_int
+from sympy.core.sympify import CantSympify
+from mpmath import isint
+from collections.abc import Sequence
```

Comment:
`collections.abc` does not seem to exist in 2.7. There is plain `abc`, but it does not contain `Sequence`. Maybe this is not necessary.